src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs

```diff
@@ -195,6 +197,45 @@ public string HubName
         /// </remarks>
         public bool RollbackEntityOperationsOnExceptions { get; set; } = true;
 
+        /// <summary>
+        /// If true, takes a lease on the task hub container, allowing for only one app to process messages in a task hub at a time.
+        /// </summary>
+        public bool UseAppLease { get; set; } = true;
+
+        /// <summary>
+        /// If UseAppLease is true, gets or sets the AppLeaaseOptions used for acquiring the lease to start the application.
+        /// </summary>
+        public AppLeaseOptions AppLeaseOptions { get; set; } = AppLeaseOptions.DefaultOptions;
+
+        internal string AppName
```

Comment:
We actually already have an AppName property defined [here](https://github.com/Azure/azure-functions-durable-extension/blob/ffa4e192b25984b9381bf32ce52fe01ff25b21e7/src/WebJobs.Extensions.DurableTask/EndToEndTraceHelper.cs#L29). Should we just reuse that so that we don't have redundant settings?

Reply:
I thought at the time that the AppName shouldnt be an empty string, but I can just use this instead.

Reply:
If the current app name is an empty string and if that isn't valid, maybe we can use `Environment.MachineName` instead when assigning a value for `AzureStorageOrchestrationServiceSettings.AppName`?

Reply:
I can't think of any Antares scenario where the environment variable wouldn't be present, but machine name seems reasonable default for those running outside of Antares.

Reply:
That's what I'm currently doing in this PR. But I did run a test to make sure an empty string was fine. It turns out it is.

Reply:
Hmm,since empty string works, My preference would be to use that AppName property.